src/uu/expr/src/locale_aware.rs

```diff
@@ -1,8 +1,25 @@
+use std::cmp::Ordering;
+
 use uucore::{
     IntoCharByteIterator,
-    i18n::{UEncoding, get_locale_encoding},
+    i18n::{
+        UEncoding,
+        collator::{AlternateHandling, CollatorOptions, locale_cmp, try_init_collator},
+        get_locale_encoding,
+    },
 };
 
+/// Perform a locale-aware string comparison using the current locale's
+/// collator.
+pub(crate) fn locale_comparison(a: &[u8], b: &[u8]) -> Ordering {
+    // Initialize the collator
+    let mut opts = CollatorOptions::default();
+    opts.alternate_handling = Some(AlternateHandling::Shifted); // This is black magic
+    let _ = try_init_collator(opts);
```

Comment:
Here, I don't exactly understand how the `CollatorOptions` work. I've just toyed with it until #5378 passed. That's why I said it's black magic.